roles/rebuild/tasks/validate.yml

```diff
@@ -2,16 +2,29 @@
   - stat:
       path: "{{ openhpc_rebuild_clouds }}"
     register: openhpc_clouds_path
-  - name: Check OpenRC file (openhpc_rebuild_clouds) exists
+  - name: Check clouds.yaml file openhpc_rebuild_clouds exists
     assert:
       that: openhpc_clouds_path.stat.exists
-      fail_msg: "openrc file {{ openhpc_rebuild_clouds }} on localhost (specified by `openhpc_rebuild_clouds`) does not exist"
-  - name: Check OpenRC file (openhpc_rebuild_clouds) is usable
-    shell: |
-      source {{ openhpc_rebuild_clouds }}
-      openstack token issue
-    changed_when: false
-    no_log: true
+      fail_msg: "clouds.yaml file {{ openhpc_rebuild_clouds }} on localhost (specified by `openhpc_rebuild_clouds`) does not exist"
+      success_msg: "{{ openhpc_rebuild_clouds }} exists"
+  - name: Read clouds.yaml # handles encrpyted files
+    include_vars:
+      file: "{{ openhpc_rebuild_clouds }}"
+      name: openhpc_rebuild_clouds_content
+    no_log: "{{ openhpc_rebuild_no_log }}"
+  - name: Check only one cloud defined
+    assert:
+      that: openhpc_rebuild_clouds_content['clouds'] | length == 1
+      fail_msg: "More than one cloud defined in {{ openhpc_rebuild_clouds }}"
+  - name: Check clouds.yaml works
+    openstack.cloud.auth:
+      # NB: can't use `cloud` parameter due to bug: https://storyboard.openstack.org/#!/story/2008600
+      auth: "{{ openhpc_rebuild_clouds_content['clouds'][cloud_name].auth }}"
+      auth_type: "{{ openhpc_rebuild_clouds_content['clouds'][cloud_name].auth_type }}"
+    vars:
+      cloud_name: "{{ openhpc_rebuild_clouds_content['clouds'] | first }}"
+    register: token
+    no_log: "{{ openhpc_rebuild_no_log }}"
+    become: yes # else get an error "Application credentials cannot request a scope."
```

Comment:
Curious - possibly picking up some environment variables and logging in as another user clears the environment?

Reply:
I guess so, I couldn't find the reason at all ...